### ISSUE TITLE
Canvas elements not captured and shown during Test Replay, shows striped placeholder

### ISSUE THREAD

**Issue by mblasco** [cypress-io/cypress#28289]
### Current behavior

Cypress tests are being executed without issues, both locally and remotely, but when we try to review the test execution in Test Replay, after some seconds, one particular canvas in the HTML rendered completely hides the test execution, preventing to see what is happening in the screen.

Inspecting the HTML during Test Replay, we found that this component:
```
<canvas data-cy-replay-is-placeholder="" data-v-25b03cc4="" class="guided-tour-canvas" width="1920" height="1080" title="Test Replay: unsupported content type" style="width: 1920px; height: 1080px;"></canvas>
```
is the one covering the whole screen.

![image](https://github.com/cypress-io/cypress/assets/1849199/74f598c4-ad1b-4276-ac7c-25a5b3dd70c1)

If I manually remove the canvas from the HTML, I can see the screen of the test execution, but as soon as I click the play to resume the execution, the canvas is added back and coverts the screen again.

This started happening some days after upgrading to Cypress 13.4.0. Reverting to Cypress 13.3.0 or upgrading to Cypress 13.5.0 is not fixing the issue. I don't think it's related to Cypress version itself, but to the cloud service that takes care of replaying the test.

### Desired behavior

That canvas should not hide test execution.

### Test code to reproduce

https://cloud.cypress.io/projects/3pahs1/runs/14754/overview/16526418-5237-4e8b-8287-f9e8e4f9f4bf/replay?att=1&roarHideRunsWithDiffGroupsAndTags=0&ts=1699515442815.4143

### Cypress Version

13.4.0

### Node version

20

### Operating System

Ubuntu

**Comment by jennifer-shehane** [comment]
@mblasco Capturing of canvas elements are not currently supported, but it's on our roadmap.

**Comment by mblasco** [comment]
I think the problem is that this canvas in our app is a transparent element whose size is the whole screen, but it's transparent, allowing to see all the elements in the screen. But from the Test Replay execution, the canvas is appearing as an opaque element covering the whole screen.

**Comment by izabelahotala** [comment]
> @mblasco Capturing of canvas elements are not currently supported, but it's on our roadmap.

Is it possible to predict when this functionality will initially work again?

**Comment by jennifer-shehane** [comment]
@mblasco Interesting, I'll relay this feedback.

**Comment by whitespacecode** [comment]
Thank you for this ticket.
We are reverting back to v12 because of this.
Our test are failing because we need to interact with the canvas and for some reason v13 can't handle it anymore?
https://cloud.cypress.io/projects/yzkmsw/runs/410/overview/5787671b-18f8-434b-85dd-6b5af96091bb/replay?roarHideRunsWithDiffGroupsAndTags=1&ts=1705310246885.3005&att=1

**Comment by jennifer-shehane** [comment]
@whitespacecode There should be no change to the interaction of canvas elements within your test. The canvas elements are simply not represented in the Test Replay. If you record a video or take screenshots of the run with canvas - you will see that it is rendered and interacted with in Cypress, so you're likely encountering another issue.

**Comment by whitespacecode** [comment]
> @whitespacecode There should be no change to the interaction of canvas elements within your test. The canvas elements are simply not represented in the Test Replay. If you record a video or take screenshots of the run with canvas - you will see that it is rendered and interacted with in Cypress, so you're likely encountering another issue.

Thank you for the reply. 
Very strange that on v13 the single test where i click the canvas failed each time. Going back to v12 fixes it.

**Comment by jennifer-shehane** [comment]
@whitespacecode Could you open a new issue with a reproducible example so we can track down the issue? It could indeed be a separate v13 related issue.

**Comment by mentaman** [comment]
@jennifer-shehane Any ETA? I see this issue exists for half year. Where can we follow it?
Unfortunately we couldn't downgrade to v12, since until we noticed this issue we already became dependent on v13 ( once we downgraded to v12 we started having a lot of flakyness )

**Comment by jennifer-shehane** [comment]
@mentaman The lack of canvas being captured on Test Replay should have zero effect on how your tests run and the output of your tests. The canvas however will not be represented in the Test Replay. If you require seeing the Canvas for review after a test is run, we recommend turning video on. 

This is still on our roadmap to provide.

**Comment by mentaman** [comment]
@jennifer-shehane I'm aware. Sorry I wasn't clear enough, but the issue for us is that our canvas is above other elements and most of the canvas is transparent in real life.
While in Replay it shows it this way instead of transparent
<img width="737" alt="image" src="https://github.com/cypress-io/cypress/assets/2057934/7ab6a54e-e6be-4c9b-bcec-3f0bf95a91f9">
so it just hides everything behind it and we can't know what happens there. and when we try to research flakyness it makes it more challenging.

If it could be transparent at least instead of showing this white/gray pattern, this would also work for us.

**Comment by varunraorao** [comment]
<img width="1396" alt="Screenshot 2024-08-20 at 9 47 12 AM" src="https://github.com/user-attachments/assets/19cb944d-1bb7-47f1-9cd2-fd341f9bcfbf">

I am facing the same issue @jennifer-shehane, this makes us difficult to verify the failed test cases during test replay mode.

**Comment by jennifer-shehane** [comment]
We're excited to announce that canvas elements can now be captured and displayed in Cypress Test Replay. When enabled, Cypress will take a screenshot representation of any canvas element and rehydrate those images within the existing live DOM during replay.

**If you'd like canvas capture enabled for your Test Replays, let us know (comment here or email support@cypress.io) —it is currently behind a feature flag that we have to enable on our side.** (We intend to have this feature flag be user facing in the future, but want to get this out as soon as possible to customers.)

To get started capturing canvas, you’ll need to be aware of a few things:
- You need to use Cypress App 13.15.0 or greater.
- Since capturing canvas elements requires additional capturing during a test run, it may affect performance, particularly if your app has many or large canvas elements or if you are on a resource constrained machine. Once enabled, we recommend monitoring for any performance issues and reaching out so we can help address them.

*Note for UI Coverage and Accessibility customers: Canvas display for UI Coverage and Accessibility is not yet available, but it’s on the way!*

**Comment by varunraorao** [comment]
Wow wonderful, Thank you @jennifer-shehane.

**Comment by varunraorao** [comment]
Thank you for taking this as priority and resolving the issue.

**Comment by nrutman** [comment]
@jennifer-shehane any update on the ability to test contents of canvas elements? Is that coming at any point in the near future?

**Comment by pjanaya** [comment]
@jennifer-shehane Thanks for the update! Any idea of when this would become publicly available?

**Comment by jennifer-shehane** [comment]
Hi, we do intend to switch Canvas capture for all recordings soon. We wanted to have it behind a custom flag initially due to some possible performance concerns, but have not materially seen any instances so far impact performance.

**Comment by pjanaya** [comment]
@jennifer-shehane thanks for the update! Is this something we could test already with a flag? I'd love to try it out!

**Comment by jennifer-shehane** [comment]
@pjanaya Yes, by emailing support@cypress.io with your organization or project id.

**Comment by henrymontoya21** [comment]
Hi @jennifer-shehane - was this flag turned on for all customers around mid-day on April 3rd? 

We have a large suite of e2e tests that covers highly interactive canvas rendering. Those tests became very flaky after experiencing 3x runtimes on April 3rd. We've been investigating possible causes and it looks like we are able to narrow down the regression to cypress cloud recording. After disabling it, our test suite appears stable and runtimes are back down. Curious if this flag could be the root cause.

**Comment by henrymontoya21** [comment]
> Hi [@jennifer-shehane](https://github.com/jennifer-shehane) - was this flag turned on for all customers around mid-day on April 3rd?
> 
 Update: nvm found the toggle https://docs.cypress.io/cloud/features/test-replay#Canvas-Elements
